dsd/constraints.py

```diff
@@ -576,6 +576,27 @@ class DomainPool(JSONSerializable):
 
     _idx: int = field(compare=False, hash=False, default=0, repr=False)
 
+    find_neighbor_sequences: bool = False
+
+    hamming_probability: Optional[Dict[int, float]] = field(default_factory=dict)
```

Comment:
I'd either make it Optional with default value None, or make it type Dict, with default value empty dict (the current default value). The advantage of the latter is that the type-checking mypy won't complain if we try to use it like a dict without first checking whether it is None.